x/mongo/driver/operation.go

```diff
@@ -627,9 +638,12 @@ func (op Operation) Execute(ctx context.Context) error {
 			}
 
 			// If the error is no longer retryable and has the NoWritesPerformed label, then we should
-			// return the previous indefinite error.
-			if tt.HasErrorLabel(NoWritesPerformed) {
-				return prevIndefiniteErr
+			// set the error to the "previous indefinite error" unless the current error is already the
+			// "previous indefinite error". After reseting, repeat the error check.
+			if tt.HasErrorLabel(NoWritesPerformed) && !errors.Is(tt, prevIndefiniteErr) {
```

Comment:
Would a variable equality check work here (e.g. `tt == prevIndefiniteErr`) instead of using `errors.Is`?

Reply:
The `WriteCommandError` struct contains a slice field, `WriteErrors`, which is not something that can be compared.  `tt != prevIndefiniteError` will result in the following compile time error:

```
# go.mongodb.org/mongo-driver/x/mongo/driver
../../../x/mongo/driver/operation.go:643:46: invalid operation: tt != prevIndefiniteErr (struct containing WriteErrors cannot be compared)
FAIL    go.mongodb.org/mongo-driver/mongo/integration/unified [build failed]
```